packages/@aws-cdk/aws-s3-notifications/lib/lambda.ts

```diff
@@ -11,19 +11,28 @@ export class LambdaDestination implements s3.IBucketNotificationDestination {
   }
 
   public bind(_scope: Construct, bucket: s3.IBucket): s3.BucketNotificationDestinationConfig {
-    const permissionId = `AllowBucketNotificationsFrom${bucket.node.uniqueId}`;
+    const permissionId = `AllowBucketNotificationsTo${this.fn.permissionsNode.uniqueId}`;
```

Comment:
Since the permission will now be added to bucket scope instead of the lambda one, we add the lambda id instead of the bucket id to preserve permission id uniqueness.